app/controllers/admin/applications_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+  
+    @pets = Pet.joins(:applications).where(applications: {id: @application.id})
```

Comment:
We don't want any of our logic to live in the controller. This would need to live in a method at the model and then you can call that method here.